### PR TITLE
[MO] - replace EmbeddedKafkaCluster with Strimzi test container

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

This PR replaces `EmbeddedKafkaCluster` with `Strimzi test container` in the user-operator, cluster-operator and topic-operator module. Thus completely removing dependency for `EmbeddedKafkaCluster`

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -581,47 +582,11 @@
...
-                <scope>test</scope>
-                <exclusions>
-                    <exclusion>
-                        <artifactId>*</artifactId>
-                        <groupId>*</groupId>
-                    </exclusion>
-                </exclusions>
+                <groupId>io.strimzi</groupId>
+                <artifactId>strimzi-test-container</artifactId>
+                <version>${strimzi-test-container.version}</version>
```
Comment (scholzj) [inline pom.xml:588]: `test` scope?

File: topic-operator/pom.xml [hunk 2]
```diff
@@ -116,26 +116,10 @@
...
-            <classifier>test</classifier>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.kafka</groupId>
-            <artifactId>kafka_2.13</artifactId>
-            <classifier>test</classifier>
-            <scope>test</scope>
+            <groupId>io.strimzi</groupId>
+            <artifactId>strimzi-test-container</artifactId>
```
Comment (scholzj) [inline topic-operator/pom.xml:121]: `test` scope?

**Review by scholzj — APPROVED** [review]
Some nits. But LGTM overall.

**Comment by tombentley** [comment]
If you want to open an issue to come back to these after release that's fine with me.

**Review by tombentley — APPROVED** [review]

**Pull request merged**